src/Scynet.HatcheryFacade/Controllers/AgentsController.cs

```diff
@@ -3,41 +3,45 @@
 using System.Linq;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
 using Orleans;
 using Scynet.GrainInterfaces.Agent;
 using Scynet.GrainInterfaces.Component;
 using Scynet.GrainInterfaces.Registry;
+using ILogger = Grpc.Core.Logging.ILogger;
 
 namespace Scynet.HatcheryFacade.Controllers
 {
     [Route("api/[controller]")]
     [ApiController]
-    public class ValuesController : ControllerBase
+    public class AgentsController : ControllerBase
     {
-        private IClusterClient ClusterClient;
+        private readonly IClusterClient ClusterClient;
+        private readonly ILogger<AgentsController> _logger;
 
-        public ValuesController(IClusterClient clusterClient)
+        public AgentsController(IClusterClient clusterClient, ILogger<AgentsController> logger)
         {
             ClusterClient = clusterClient;
-            Console.WriteLine(clusterClient);
+            _logger = logger;
         }
 
-        // GET api/values
+        // GET api/agents
         [HttpGet]
-        public async Task<ActionResult<IEnumerable<string>>> Get()
+        public async Task<ActionResult<IDictionary<string, List<KeyValuePair<Guid, AgentInfo>>>>> Get()
         {
             // HACK: testing code below
             var registry = ClusterClient.GetGrain<IRegistry<Guid, AgentInfo>>(0);
             //await registry.Register(Guid.Empty, new AgentInfo { });
             //await registry.Register(Guid.NewGuid(), new AgentInfo { });
             //await registry.Register(Guid.NewGuid(), new AgentInfo { });
 
-            var res = await registry.Query(x =>
-                from y in x
-                select y);
+            var agentGroup = await registry.Query(agents =>
+                from agent in agents
+                group agent by agent.Value.RunnerType into category
+                select Tuple.Create(category.Key, category.ToList()));
 
 
-            return res.Select(pair => pair.Key.ToString()).ToArray();
+            return agentGroup.ToDictionary(category => category.Item1, category => category.Item2);
```

Comment:
You can put the `ToDictionary` call inside the call to Query, if that would be simpler to write.